File: backend/models/log_entry.py
class LogEntry:
    """
    LogEntry — Model representing a single network log event.

    Fields:
      - id          : Database row ID (0 or None if not persisted)
      - time        : Timestamp when the event was recorded
      - ip          : Source IP address
      - action      : Event type (LOGIN_FAIL, LOGIN_SUCCESS, REQUEST)
      - score       : Risk score computed by SecurityBot (0–100, default 0)
      - status      : Status label (PASS, SUSPICIOUS, MONITORING, BLOCKED)
      - attack_type : Attack classification (NORMAL, BRUTE_FORCE, REQUEST_FLOOD)
      - description : Optional detail string
      - is_modified : Track if score/status/attack_type changed after creation/load
    """

    def __init__(self, time: str, ip: str, action: str, id_val: int = 0):
        self.id = id_val
        self.time = time
        self.ip = ip
        self.action = action
        self.score = 0
        self.status = "PASS"
        self.attack_type = "NORMAL"
        self.description = None
        self.is_modified = False

    def is_modified_flag(self) -> bool:
        return self.is_modified

    def get_id(self) -> int:
        return self.id

    def set_id(self, id_val: int):
        self.id = id_val

    def get_time(self) -> str:
        return self.time

    def get_ip(self) -> str:
        return self.ip

    def get_action(self) -> str:
        return self.action

    def get_score(self) -> int:
        return self.score

    def set_score(self, score: int):
        if self.score != score:
            self.score = score
            self.is_modified = True

    def get_status(self) -> str:
        return self.status

    def set_status(self, status: str):
        if self.status != status:
            self.status = status
            self.is_modified = True

    def get_attack_type(self) -> str:
        return self.attack_type

    def set_attack_type(self, attack_type: str):
        if self.attack_type != attack_type:
            self.attack_type = attack_type
            self.is_modified = True

    def get_description(self) -> str:
        return self.description

    def set_description(self, description: str):
        self.description = description

    def reset_modified(self):
        self.is_modified = False

    def to_dict(self) -> dict:
        return {
            "id": self.id,
            "time": self.time,
            "ip": self.ip,
            "action": self.action,
            "attack": self.attack_type,
            "score": self.score,
            "status": self.status,
            "description": self.description
        }

    def __repr__(self) -> str:
        return (f"[{self.time}] {self.ip} {self.action} | "
                f"score={self.score} | status={self.status} | attack={self.attack_type}")

File: archive/java_src/model/LogEntry.java
package model;

/**
 * LogEntry — Model representing a single network log event.
 *
 * Fields:
 *   - time       : Timestamp when the event was recorded
 *   - ip         : Source IP address
 *   - action     : Event type (LOGIN_FAIL, LOGIN_SUCCESS, REQUEST)
 *   - score      : Risk score computed by SecurityBot (0–100, default 0)
 *   - status     : Status label (PASS, SUSPICIOUS, MONITORING, BLOCKED)
 *   - attackType : Attack classification (NORMAL, BRUTE_FORCE, REQUEST_FLOOD)
 *
 * score, status, and attackType are assigned by SecurityBot AFTER analysis.
 */
public class LogEntry {

    /** Database row ID (0 if not yet persisted) */
    private int id;

    /** Timestamp of the log event */
    private String time;

    /** Source IP address */
    private String ip;

    /** Recorded action (LOGIN_FAIL, LOGIN_SUCCESS, REQUEST) */
    private String action;

    /** Risk score (0–100); computed by SecurityBot — default: 0 */
    private int score;

    /** Status after analysis — default: "PASS" */
    private String status;

    /** Detected attack type — default: "NORMAL" */
    private String attackType;

    /** Optional detailed description (from database, may be null) */
    private String description;

    /** Flag indicating whether properties were modified after loaded */
    private boolean isModified;

    /**
     * Creates a LogEntry with basic data from a log file or generator.
     * score / status / attackType will be set by SecurityBot later.
     *
     * @param time   Timestamp (e.g. ISO format or "HH:mm:ss")
     * @param ip     Source IP address (e.g. "192.168.1.15")
     * @param action Event type (e.g. "LOGIN_FAIL")
     */
    public LogEntry(String time, String ip, String action) {
        this.time = time;
        this.ip = ip;
        this.action = action;
        this.score = 0;
        this.status = "PASS";
        this.attackType = "NORMAL";
        this.isModified = false;
    }

    // ===================== GETTERS =====================

    /** @return true if log has been modified by bot since creation */
    public boolean isModified() { return isModified; }

    /** @return Database row ID */
    public int getId() { return id; }

    /** @return Timestamp string */
    public String getTime() { return time; }

    /** @return Source IP address */
    public String getIp() { return ip; }

    /** @return Action type (LOGIN_FAIL / LOGIN_SUCCESS / REQUEST) */
    public String getAction() { return action; }

    /** @return Risk score (0–100) */
    public int getScore() { return score; }

    /** @return Current status (PASS / SUSPICIOUS / MONITORING / BLOCKED) */
    public String getStatus() { return status; }

    /** @return Detected attack type (NORMAL / BRUTE_FORCE / REQUEST_FLOOD) */
    public String getAttackType() { return attackType; }

    /** @return Optional description (may be null) */
    public String getDescription() { return description; }

    // ===================== SETTERS =====================

    /** Sets the risk score (called by SecurityBot). */
    public void setScore(int score) { 
        if (this.score != score) {
            this.score = score;
            this.isModified = true;
        }
    }

    /** Sets the status label (called by SecurityBot). */
    public void setStatus(String status) { 
        if (this.status == null || !this.status.equals(status)) {
            this.status = status;
            this.isModified = true;
        }
    }

    /** Sets the attack type (called by SecurityBot). */
    public void setAttackType(String attackType) { 
        if (this.attackType == null || !this.attackType.equals(attackType)) {
            this.attackType = attackType;
            this.isModified = true;
        }
    }

    /** Sets the database row ID. */
    public void setId(int id) { this.id = id; }

    /** Sets the optional description. */
    public void setDescription(String description) { this.description = description; }

    /**
     * Resets the isModified flag to false.
     * Called after loading a LogEntry from DB so that updateLogs() only writes
     * back entries that SecurityBot actually modified during analysis,
     * not every log on every cycle.
     */
    public void resetModified() { this.isModified = false; }

    /**
     * Returns a string representation for debugging.
     */
    @Override
    public String toString() {
        return "[" + time + "] " + ip + " " + action
                + " | score=" + score
                + " | status=" + status
                + " | attack=" + attackType;
    }
}